client_end_user/shared/src/commonMain/kotlin/presentation/home/HomeScreen.kt

```diff
@@ -133,11 +139,14 @@ class HomeScreen : BaseScreen<
 
             item {
                 BpSimpleTextField(
-                    "",
+                    text = state.searchTerm,
                     hint = Resources.strings.searchHint,
                     hintColor = Theme.colors.contentSecondary,
-                    onValueChange = {},
-                    onClick = { listener.onClickSearch() },
+                    onValueChange = listener::onChangeSearchText,
+                    onClick = {
+                        listener.onClickSearch()
+                        tabNavigator.current = SearchTab
+                              },
                     leadingPainter = painterResource(Resources.images.searchOutlined),
                     modifier = Modifier.padding(horizontal = 16.dp)
                 )
```

Comment:
```
                BpSimpleTextField(
                    text = state.searchTerm,
                    hint = Resources.strings.searchHint,
                    hintColor = Theme.colors.contentSecondary,
                    enabled = false, // we need to pass enabled to text field with value false
                    onValueChange = listener::onChangeSearchText,
                    leadingPainter = painterResource(Resources.images.searchOutlined),
                    modifier = Modifier.padding(horizontal = 16.dp)
                        .noRippleEffect { tabNavigator.current = SearchTab } // use noRipple modifier directly to the hole parent
                )
```

Reply:
i need to click icon and it use noRippleEffect in texst already  the modifier is for all text not just icon i don't think i need to add on click on whole text